protos/flyteidl/event/event.proto

```diff
@@ -151,12 +151,30 @@ message TaskExecutionEvent {
     // The version field should be incremented when metadata changes across the duration of an individual phase.
     uint32 phase_version                  = 12;    
 
+    // If there is an explanation for this phase transition, the reason will capture it.
+    string reason                         = 13;
+
+
+    // A predefined yet extensible Task type identifier. If the task definition is already registered in flyte admin
+    // this type will be identical, but not all task executions necessarily use pre-registered definitions and this
+    // type is useful to render the task in the UI, filter task executions, etc.
+    string type                           = 14;
+
     // Metadata around how a task was executed.
     TaskExecutionMetadata metadata        = 16;
 }
 
+
+// This message holds task execution metadata specific to resource allocation used to manage concurrent
+// executions for a project namespace.
+message ManagedResourceInfo {
+    // Unique resource ID used to identify this execution when allocating a token.
+    string token                         = 1;
+}
+
 // Holds metadata around how a task was executed.
 // TODO(katrogan): Extend to include freeform fields (https://github.com/flyteorg/flyte/issues/325).
+// Next ID to use:  3
```

Comment:
While I liked these back in the day, unless we have tools to make sure they are correct (do we?) they will get out of sync and be confusing... I think.

Reply:
fair point, done